File: aula_05/src/main/kotlin/cleanarch/domain/CarPlate.kt
package com.branas.cleanarch.domain

private const val VALID_CAR_PLATE_REGEX = "^[A-Za-z]{3}[0-9]{4}"

data class CarPlate(val value: String) {
    init {
        if (!this.validate()) throw Exception("Invalid car plate")
    }

    private fun validate(): Boolean {
        return this.value.isNotBlank() && this.value.matches(VALID_CAR_PLATE_REGEX.toRegex())
    }
}

File: aula_06/frontend/src/domain/CarPlate.js
export default class CarPlate {

	constructor (value) {
		if (!this.validate(value)) throw new Error("Invalid car plate");
		this.value = value;
	}

	validate (email) {
		return String(email)
			.toLowerCase()
			.match(
			/^[a-z]{3}[0-9]{4}$/
			);
	}

	getValue () {
		return this.value;
	}
}